unit/java_bytecode/java_bytecode_convert_method/convert_initalizers.cpp

```diff
@@ -0,0 +1,250 @@
+/*******************************************************************\
+
+ Module: Unit tests for converting abstract classes
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+
+#include <memory>
+
+#include <util/config.h>
+#include <util/symbol_table.h>
+
+#include <java_bytecode/java_bytecode_language.h>
+
+#include <testing-utils/load_java_class.h>
+#include <testing-utils/require_type.h>
+
+struct test_datat
+{
+  symbol_tablet symbol_table;
+  std::string constructor_descriptor;
+};
+
+/// Verify that a given descriptor is marked as a constructor in the symbol
+/// table
+/// \param test_data The data to run the test on
+void require_is_constructor(const test_datat &test_data)
+{
+
+  const symbolt &constructor =
+    test_data.symbol_table.lookup_ref(test_data.constructor_descriptor);
+  THEN("The constructor should be marked as a constructor")
+  {
+    code_typet constructor_type = require_type::require_code(constructor.type);
+    REQUIRE(constructor_type.get_is_constructor());
+  }
+}
+
+/// Verify that a given descriptor is not marked as a constructor in the symbol
+/// table
+/// \param test_data The data to run the test on
+void require_is_static_initalizer(const test_datat &test_data)
```

Comment:
Typo: initializer

Reply:
This function checks that it is not a constructor, but does not verify that it is static initializer. I would either add check for `clinit` in the name or rename the function.